cmake/Modules/GrPython.cmake

```diff
@@ -93,12 +93,12 @@ endmacro(GR_PYTHON_CHECK_MODULE)
 ########################################################################
 if(NOT DEFINED GR_PYTHON_DIR)
 execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "
-import os
-import sys
-if os.name == 'posix':
-    print(os.path.join('lib', 'python' + sys.version[:3], 'dist-packages'))
-if os.name == 'nt':
-    print(os.path.join('Lib', 'site-packages'))
+try:
+    from site import getsitepackages
+    print(getsitepackages()[0])
+except ImportError:
```

Comment:
under which circumstances can this fail? isn't `site` part of Python 2.7+?

Reply:
The `getsitepackages` is missing from `virtualenv` (https://github.com/pypa/virtualenv/issues/228).